oak/server/storage/storage_node.cc

```diff
@@ -31,137 +31,164 @@ StorageNode::StorageNode(BaseRuntime* runtime, const std::string& name,
                          const std::string& storage_address)
     : NodeThread(runtime, name), storage_processor_(storage_address) {}
 
-void StorageNode::Run(Handle request_handle) {
+void StorageNode::Run(Handle invocation_handle) {
   // Borrow a pointer to the relevant channel half.
-  MessageChannelReadHalf* request_channel = BorrowReadChannel(request_handle);
-  if (request_channel == nullptr) {
-    LOG(ERROR) << "Required channel not available; handle: " << request_handle;
+  MessageChannelReadHalf* invocation_channel = BorrowReadChannel(invocation_handle);
+  if (invocation_channel == nullptr) {
+    LOG(ERROR) << "Required channel not available; handle: " << invocation_handle;
     return;
   }
   std::vector<std::unique_ptr<ChannelStatus>> channel_status;
-  channel_status.push_back(absl::make_unique<ChannelStatus>(request_handle));
+  channel_status.push_back(absl::make_unique<ChannelStatus>(invocation_handle));
   while (true) {
     if (!WaitOnChannels(&channel_status)) {
       LOG(WARNING) << "Node termination requested";
       return;
     }
 
-    ReadResult result = request_channel->Read(INT_MAX, INT_MAX);
-    if (result.required_size > 0) {
-      LOG(ERROR) << "Message size too large: " << result.required_size;
+    // Expect to receive a pair of handles in the invocation message:
+    //  - Handle to the read half of a channel that holds the request, serialized
+    //    as a GrpcRequest.
+    //  - Handle to the write half of a channel to send responses down, each
+    //    serialized as a GrpcResponse.
+    ReadResult invocation = invocation_channel->Read(INT_MAX, INT_MAX);
```

Comment:
It seems it would be worth defining a wrapper type for this, similar to https://github.com/project-oak/oak/blob/master/sdk/rust/oak/src/grpc/invocation.rs . That said, I am not sure how long this code will survive before being thrown away / rewritten anyways, so not sure it's worth it.

Reply:
Done.